common/thorhelper/thorsoapcall.cpp

```diff
@@ -1921,11 +1935,18 @@ class CWSCAsyncFor : implements IWSCAsyncFor, public CInterface, public CAsyncFo
             ForEach(*iter)
             {
                 const char * key = iter->getPropKey();
-                if (!httpHeaderBlockContainsHeader(httpheaders, key))
+                bool hasHeader = httpHeaderBlockContainsHeader(httpheaders, key);
+                if (!hasHeader)
                 {
-                    const char * value = iter->queryPropValue();
-                    if (!isEmptyString(value))
-                        request.append(key).append(": ").append(value).append("\r\n");
+                    //If this header is http-global-id, check that global-id hasn't been explicitly added already
+                    const char * altHeader = queryAlternativeHeader(key);
```

Comment:
If a query set "hpcc-global-id" but the headers line set "global-id" or vice-versa you would get both values in the headers potentially causing confusion.  Only likely to affect the regression tests - but it did once the ids were preserved where they were previously dropped.